connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SessionHolder.scala

```diff
@@ -47,6 +48,9 @@ case class SessionHolder(userId: String, sessionId: String, session: SparkSessio
   // foreachBatch() in Streaming. Lazy since most sessions don't need it.
   private lazy val dataFrameCache: ConcurrentMap[String, DataFrame] = new ConcurrentHashMap()
 
+  private lazy val listenerCache: ConcurrentMap[String, StreamingQueryListener] =
```

Comment:
Should this be part of the StreamingQuery itself? It does not need to be part of the session.
Also, when is this removed?

Reply:
Nm. I wanted to remove this comment. It is part of the StreamingQueryManager, not individual queries. 
Yes, we need to fix removing the listener.